src/components/container/Goal/Goal.tsx

```diff
@@ -0,0 +1,103 @@
+    const inProgressColor = resolveColor(layoutContext?.colorScheme, props.inProgressColor) ?? 'var(--mdhui-color-primary)';
+    const completedColor = resolveColor(layoutContext?.colorScheme, props.completedColor) ?? 'var(--mdhui-color-success)';
+
+    useInitializeView(() => {
+        setLoading(true);
+        setValue(undefined);
+
+        if (props.previewState === 'loading') {
+            return;
+        }
+
+        const valueProvider = props.previewState
+            ? createPreviewValueProvider(props.previewState, props.targetValue, props.maxValue)
+            : props.valueProvider;
+
+        valueProvider.getValue().then(value => {
+            setValue(Math.max(0, Math.min(value ?? 0, props.maxValue)));
+            setLoading(false);
+        })
+    }, [], [props.previewState, props.targetValue, props.maxValue, props.valueProvider]);
+
+    const getSegmentColor = (index: number) => {
+        let segmentColor = notStartedColor;
+        if (value !== undefined && index <= value) {
+            segmentColor = value >= props.targetValue ? completedColor : inProgressColor;
+        }
+        return segmentColor;
+    };
+
+    let statusColor = notStartedColor;
+    if (value !== undefined && value > 0) {
+        statusColor = value >= props.targetValue ? completedColor : inProgressColor;
+    }
+
+    const data: { value: number, color: string }[] = [];
+    for (let i = 1; i <= props.maxValue; i++) {
+        data.push({ value: 1, color: getSegmentColor(i) });
+    }
+
+    return <div className="mdhui-goal" ref={props.innerRef}>
+        <div className="mdhui-goal-chart">
+            <ResponsiveContainer>
+                <PieChart>
+                    <Pie
+                        data={data} dataKey="value"
+                        cx="50%" cy="50%"
+                        innerRadius={28} outerRadius={40}
+                        startAngle={90} endAngle={450} paddingAngle={5}
+                        isAnimationActive={false}
+                        strokeWidth={0}
+                    >
+                        {data.map((entry, index) => <Cell key={index} style={{ fill: entry.color }} />)}
+                    </Pie>
+                </PieChart>
+            </ResponsiveContainer>
+            <div className="mdhui-goal-chart-icon-wrapper">
+                {<FontAwesomeIcon icon={props.icon ?? faStar} className="mdhui-goal-chart-icon" style={{ color: statusColor }} />}
+            </div>
+        </div>
+        <div className="mdhui-goal-info">
+            {loading &&
+                <div className="mdhui-goal-ratio-loading">
+                    <FontAwesomeSvgIcon icon={faArrowsRotate} spin style={{ color: notStartedColor }} />
+                </div>
+            }
+            {!loading &&
+                <div className="mdhui-goal-ratio" style={{ color: statusColor }}>{value + '/' + props.maxValue}</div>
+            }
+            <div className="mdhui-goal-name">{props.label}</div>
```

Comment:
this class is not in the .css

Reply:
I like to give most all `div` elements a style class even if I haven't had to add any CSS rules yet, just in case we end up needing to add some or if a user of this component needs to override something with CSS rules downstream.  If not present, the CSS selector for the `div` becomes more complex and fragile (i.e. something like `.mdhui-goal-info > div:last-child` instead of just `.mdhui-goal-name`).